ibis/expr/window.py

```diff
@@ -31,10 +29,43 @@ def _determine_how(preceding):
             'Type {} is not supported for row- or range- based trailing '
             'window operations'.format(offset_type)
         )
-
     return how
 
 
+@functools.singledispatch
+def _get_preceding_value(preceding):
+    raise TypeError(
+        "Type {} is not a valid type for 'preceding' "
+        "parameter".format(type(preceding))
+    )
+
+
+@_get_preceding_value.register(tuple)
+def _get_preceding_value_tuple(preceding):
+    start, end = preceding
+    if start is None:
+        preceding_value = end
+    else:
+        preceding_value = start
+    return preceding_value
+
+
+@_get_preceding_value.register(int)
+@_get_preceding_value.register(np.integer)
+@_get_preceding_value.register(ir.IntervalScalar)
+def _get_preceding_value_simple(preceding):
+    return preceding
+
+
+@_get_preceding_value.register(RowsWithMaxLookback)
+def _get_preceding_value_mlb(preceding):
+    preceding_value = preceding.rows
+    if not isinstance(preceding_value, int):
```

Comment:
This could be an `np.integer` as well can you add that to the list of types to check here?

Reply:
Yes, added in next commit